internal/k8s/epslices/endpoint_slices.go

```diff
@@ -11,16 +11,13 @@ import (
 
 const SlicesServiceIndexName = "ServiceName"
 
-// IsConditionServing tells if the conditions represent a serving state, deferring
-// to ready state if serving == nil.
-func IsConditionServing(conditions discovery.EndpointConditions) bool {
-	if conditions.Serving == nil {
-		if conditions.Ready == nil {
-			return true
-		}
-		return *conditions.Ready
+// IsConditionReady tells if the conditions represent a ready state, deferring
+// to serving state if ready == false.
+func IsConditionReady(conditions discovery.EndpointConditions) bool {
```

Comment:
given this is going to be a bit more complex, can we add a unit tests to cover the various cases?